### ISSUE TITLE
[Bug] NPE during compilation (getRawMethodFromCache)

### ISSUE THREAD

**Issue by koxx12-dev** [manifold-systems/manifold#545]
**Describe the bug**
Working on a minecraft mod and i wanted to use manifold to make my code concise and readable, but after adding it every build attempt resulted in a NPE during `compileJava`. Tried solution in #65 just to make sure it wasn't it and it didn't help

**To Reproduce**
1. Clone https://github.com/koxx12-dev/Example-Forge-1.8.9 (main branch not kotlin, make sure to read the readme)
2. Add mainfold-ext, manifold-ext-rt and `-Xplugin:Manifold` compilerArg
3. Run gradle build task
4. java.lang.NullPointerException

**Expected behavior**
Compile Successfully and not throw a NPE on every attempted build

**Screenshots**
![idea64_yXJF0zlu89](https://github.com/manifold-systems/manifold/assets/80456351/e193d23f-e2c0-4609-af14-0ac20eeaaa8e)

**Desktop (please complete the following information):**
- OS Type & Version: Windows 10 22H2
- Java/JDK version: (compile) 1.8.0_382 Azul Zulu / (gradle) 17.0.9 Terumin
- IDE version (IntelliJ IDEA or Android Studio): IntelliJ IDEA 2023.3.4 (Ultimate Edition)
- Manifold version: 2024.1.1 (both manifold-ext and mainfold-ext-rt)
- Manifold IntelliJ plugin version:  2023.3.15

**Additional context**
Its always happening with my template, also sorry if i made any mistakes its my first time making a serious issue

**Stack trace**
```
An exception has occurred in the compiler (1.8.0_382). Please file a bug against the Java compiler via the Java bug reporting page (http://bugreport.java.com) after checking the Bug Database (http://bugs.java.com) for duplicates. Include your program and the following diagnostic in your report. Thank you.
java.lang.RuntimeException: java.lang.NullPointerException
	at com.sun.tools.javac.main.Main.compile(Main.java:473)
	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
	at org.gradle.internal.compiler.java.IncrementalCompileTask.call(IncrementalCompileTask.java:92)
	at org.gradle.api.internal.tasks.compile.AnnotationProcessingCompileTask.call(AnnotationProcessingCompileTask.java:94)
	at org.gradle.api.internal.tasks.compile.ResourceCleaningCompilationTask.call(ResourceCleaningCompilationTask.java:57)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:55)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:39)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractIsolatedCompilerWorkerExecutor$CompilerWorkAction.execute(AbstractIsolatedCompilerWorkerExecutor.java:78)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:54)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:48)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
	at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:48)
	at org.gradle.workers.internal.FlatClassLoaderWorker.run(FlatClassLoaderWorker.java:32)
	at org.gradle.workers.internal.FlatClassLoaderWorker.run(FlatClassLoaderWorker.java:22)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:96)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:65)
	at org.gradle.process.internal.worker.request.WorkerAction$1.call(WorkerAction.java:138)
	at org.gradle.process.internal.worker.child.WorkerLogEventListener.withWorkerLoggingProtocol(WorkerLogEventListener.java:41)
	at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:135)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.NullPointerException
	at manifold.util.concurrent.ConcurrentWeakHashMap.get(ConcurrentWeakHashMap.java:1009)
Caused by: java.lang.NullPointerException

	at manifold.util.ReflectUtil.getRawMethodFromCache(ReflectUtil.java:1233)
	at manifold.util.ReflectUtil.getMethodFromCache(ReflectUtil.java:1223)
	at manifold.util.ReflectUtil._method(ReflectUtil.java:327)
	at manifold.util.ReflectUtil.method(ReflectUtil.java:317)
	at manifold.util.ReflectUtil.method(ReflectUtil.java:300)
	at manifold.internal.javac.JavacPlugin.overrideJavacToolEnter(JavacPlugin.java:206)
	at manifold.internal.javac.JavacPlugin.init(JavacPlugin.java:198)
	at manifold.internal.javac.JavacPluginBootstrap.init(JavacPluginBootstrap.java:49)
	at com.sun.tools.javac.main.Main.compile(Main.java:470)
	... 34 more
```

**Comment by rsmckinney** [comment]
Hi @koxx12-dev.

I've modified your `build.gradle.kts` file to include manifold-ext etc. Builds okay on my end.

```kts
import dev.deftu.gradle.utils.GameSide

plugins {
    java
    val dgtVersion = "1.22.4"
    id("dev.deftu.gradle.tools") version(dgtVersion)
    id("dev.deftu.gradle.tools.shadow") version(dgtVersion)
    id("dev.deftu.gradle.tools.blossom") version(dgtVersion)
    id("dev.deftu.gradle.tools.minecraft.loom") version(dgtVersion)
    id("dev.deftu.gradle.tools.resources") version(dgtVersion)
}

toolkit.useDevAuth()

toolkitLoomHelper  {
    useForgeMixin(modData.id)
    useMixinRefMap(modData.id)
    useTweaker("org.spongepowered.asm.launch.MixinTweaker")
    useProperty("mixin.debug.export", "true", GameSide.CLIENT)
    disableRunConfigs(GameSide.SERVER)
}

repositories {
    maven("https://repo.spongepowered.org/maven/")
    maven("https://maven.deftu.dev/releases")
    mavenCentral()
}

java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(17))
    }
}

tasks.compileJava {
    options.release.set(8)
}

val manifoldVersion = "2024.1.1"

dependencies {
    implementation( "systems.manifold:manifold-ext-rt:$manifoldVersion")
    annotationProcessor("systems.manifold:manifold-ext:$manifoldVersion")

    implementation(shade("org.spongepowered:mixin:0.7.11-SNAPSHOT") {
        isTransitive = false
    })
}

tasks.withType<JavaCompile> {
    options.compilerArgs.addAll(listOf("-Xplugin:Manifold"))
}
```

**Comment by rsmckinney** [comment]
Ah... I think I found the problem you are experiencing.

You probably want Java 8 for your JDK, instead of using JDK 17 to compile for Java 8:

```kts
java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(8))
    }
}
```

In that case I see the NPE you found. I'm working on that now. Should have a fix ready by EOD tomorrow. Thanks for reporting this.

**Comment by rsmckinney** [comment]
Fix available with release `2024.1.2`

I used the following build.gradle.kts with a JDK 8 toolchain.

```kts
import dev.deftu.gradle.utils.GameSide

plugins {
    java
    val dgtVersion = "1.22.4"
    id("dev.deftu.gradle.tools") version(dgtVersion)
    id("dev.deftu.gradle.tools.shadow") version(dgtVersion)
    id("dev.deftu.gradle.tools.blossom") version(dgtVersion)
    id("dev.deftu.gradle.tools.minecraft.loom") version(dgtVersion)
    id("dev.deftu.gradle.tools.resources") version(dgtVersion)
}

toolkit.useDevAuth()

toolkitLoomHelper  {
    useForgeMixin(modData.id)
    useMixinRefMap(modData.id)
    useTweaker("org.spongepowered.asm.launch.MixinTweaker")
    useProperty("mixin.debug.export", "true", GameSide.CLIENT)
    disableRunConfigs(GameSide.SERVER)
}

repositories {
    maven("https://repo.spongepowered.org/maven/")
    maven("https://maven.deftu.dev/releases")
    mavenCentral()
}

java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(17))
    }
}

tasks.compileJava {
    options.release.set(8)
}

val manifoldVersion = "2024.1.1"

dependencies {
    implementation( "systems.manifold:manifold-ext-rt:$manifoldVersion")
    annotationProcessor("systems.manifold:manifold-ext:$manifoldVersion")

    implementation(shade("org.spongepowered:mixin:0.7.11-SNAPSHOT") {
        isTransitive = false
    })
}

tasks.withType<JavaCompile> {
    options.compilerArgs.addAll(listOf("-Xplugin:Manifold"))
}```

**Comment by koxx12-dev** [comment]
Sorry for responding so late, i was trying to find a workaround around this issue which i found right before you pushed out the proper fix (and also break the entire build setup in the process). Nonetheless after updating to 2024.1.2 everything works perfectly with the current build script, so i would conciser this issue solved
